src/Psy/Readline/TabCompletion.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace Psy\Readline;
+
+use Psy\Context;
+
+/**
+ * Class TabCompletion
+ * @package Psy\Readline
+ */
+class TabCompletion
+{
+    /** @var Context */
+    protected $context;
+
+    public function __construct(Context $context = null)
+    {
+        $this->context = $context;
+    }
+
+    /**
+     * @param Context $context
+     */
+    public function setContext(Context $context)
+    {
+        $this->context = $context;
+    }
+
+
+    public function activate()
+    {
+        readline_completion_function(array(&$this, 'callback'));
+    }
+
+    /**
+     * @param $input
+     * @param $index
+     * @return array
+     */
+    public function callback($input, $index)
```

Comment:
I'd love to be able to test this... Can we extract the guts of `callback()` into a function that takes `$input, $index, readline_info())` so that we can pass in fake data?